compiler/semexprs.nim

```diff
@@ -2521,6 +2521,11 @@ proc semSetConstr(c: PContext, n: PNode, expectedType: PType = nil): PNode =
       typ = makeRangeType(c, 0, MaxSetElements-1, n.info)
     elif lengthOrd(c.config, typ) > MaxSetElements:
       typ = makeRangeType(c, 0, MaxSetElements-1, n.info)
+    if expectedElementType.sym != nil and 
+        sfImportc notin expectedElementType.sym.flags and
+        lengthOrd(c.config, expectedElementType) > MaxSetElements and
+        not (expectedElementType.isIntLit and expectedElementType.n.intVal <= MaxSetElements):
+      localError(c.config, n.info, errSetTooBig)
```

Comment:
This rule makes no sense. The real check seems to be `lengthOrd(c.config, expectedElementType) > MaxSetElements`